src/compiler/grammar/Shexl.g4

```diff
@@ -0,0 +1,270 @@
+ | NOT_KW triple_expression
+ | node_constraint
+ | '{' triple_constraint '}'                            // A single triple constraint.
+ | '{' (triple_constraint ';')+ triple_constraint '}'   // Multiple triple constraints. (eachOfs)
+ ;
+
+// A triple constraint is the abstraction of the well known
+// eachOf of ShEx. In our case as we don't have the eachOneOf
+// we don't need to expand this rule. Therefore the triple
+// constraint is defined as a property, a node constraint and
+// the cardinality. If no cardilaity is pressent the default one
+// will be the [0,n).
+triple_constraint
+ : prefix_invocation node_constraint cardinality?
+ ;
+
+// A prefix invocation is the invocation to something declared
+// in the prefix. For example when we declare a property name
+// in a triple constraint we invoque the property associated to
+// the declared prefix. And the same with Datatypes.
+prefix_invocation
+ : prefix=LABEL? ':' property=LABEL
+ ;
+
+// Describe the allow values of a node. It can be:
+//  - anything.
+//  - a datatype. Notice that a datatype actually is a prefix
+//      invocation.
+//  - a node kind.
+//  - a value set.
+//  - or a shape reference.
+node_constraint
+ : '.'                  // Anything.
+ | prefix_invocation    // Datatype.
+ | LITERAL_KW           // Node kind.
+ | IRI_KW
+ | BNODE_KW
+ | NON_LITERAL_KW
+ | '[' value_set_type* ']'  // Value set.
+ | shape_invocation     // Shape reference.
+ ;
+
+// Represents the possible values that a value set might contain.
+value_set_type
+ : prefix_invocation    // Prefix invocation.
+ | shape_invocation     // Shape invocation.
+ | STRING_LITERAL       // String literal.
+ | REAL_LITERAL         // Real literal.
+ ;
+
+// As seen previously a triple constraint is formed by a property, a
+// node constraint and a cardinality. The cardinality might be take
+// different values depending on wat does it means:
+//  - `*` -> any number of repetitions, 0 or more. If no cardinality
+//           is set this one is de default one. [0,n)
+//  - `+` -> any number of repetitions but at least 1. [1,n)
+//  - `?` -> none or one appearance. Also known as optional. [0,1]
+//  - `{N}` -> exactly N repetitions [N,N].
+//  - `{N,M}` -> a minimum of N and a maximum of M repetitions. [N,M]
+//  - `{N, }` -> any number of repetitions but a minimum of N. [N,m)
+cardinality
+ : '*'
+ | '+'
+ | '?'
+ | '{' INT_LITERAL '}'
+ | '{' INT_LITERAL ',' INT_LITERAL '}'
+ | '{' INT_LITERAL ',''}'
+ ;
+
+// TOKENS
+
+PREFIX_KW       :   'PREFIX'        ;
+BASE_KW         :   'BASE'          ;
+IMPORT_KW       :   'IMPORT'        ;
+START_KW        :   'START'         ;
+AND_KW          :   'AND'           ;
+OR_KW           :   'OR'            ;
+NOT_KW          :   'NOT'           ;
+IRI_KW          :   'IRI'           ;
+LITERAL_KW      :   'LITERAL'       ;
+BNODE_KW        :   'BNODE'         ;
+NON_LITERAL_KW  :   'NONLITERAL'    ;
+
+
+LABEL
+ : [a-zA-Z_][a-zA-Z0-9_]*
+ ;
```

Comment:
ShEx is more flexible when it comes to allowed characters in labels. For example, something like this will work:
```ttl
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX sch-ema: <http://schema.org/>

<User> {
 sch-ema:1n-ame  xsd:string ;
 sch-ema:knows @<User>*
}
```

The same happens in ttl for example. I think it is worth following the already stablished conventions, but I would also like to receive your feedback on this topic.

Reply:
Fixed at 69127fcf95ab4af221236069ec8a1079d9376d41.